DataFormats/Common/interface/WrapperBase.h

```diff
@@ -51,12 +51,12 @@ namespace edm {
     bool isProductEqual(WrapperBase const* newProduct) const { return isProductEqual_(newProduct); }
     bool hasSwap() const { return hasSwap_(); }
     void swapProduct(WrapperBase* newProduct) { swapProduct_(newProduct); }
+    void moveFrom(void* ptr, std::type_info const& type) { moveFrom_(ptr, type); }
```

Comment:
I've split `moveFrom()`/`moveFrom_()` for consistency with the other methods, though I.m not sure why this is being done...

Reply:
I believe this is where this pattern came from. I am not advocating this. I have implemented things this way sometimes just because long long ago I was told to do things this way... 

http://www.gotw.ca/publications/mill18.htm

This NVI pattern is not in CMS coding guidelines. There is the requirement "Do not inline virtual functions", but that is a little different.

If we are not following this pattern anymore, it would be nice to know.